### PR TITLE
Add kcp workspace support to two controller pipeline and snapshot

### PR DESCRIPTION
**Pull request by hongweiliu17**
Add kcp workspace to controller to pipeline and snapshot
Add samples to run integration service on kcp workspace

JIRA: HACBS-713

Signed-off-by: Hongwei Liu <hongliu@redhat.com>

### CODE REVIEW

File: config/kcp/samples/apibinding_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: apis.kcp.dev/v1alpha1
+kind: APIBinding
+metadata:
+  name: kubernetes
+spec:
+  reference:
+    workspace:
+      path: root:users:xr:vd:rh-sso-hongweiliu:hongliu-dev-7
```
Comment (jsztuka) [inline config/kcp/samples/apibinding_integration.yaml:8]: I suppose these need to change

Comment (hongweiliu17) [reply config/kcp/samples/apibinding_integration.yaml]: Yes, updated. Thanks!

**Review by Dannyb48 — APPROVED** [review]
LGTM

File: syncer710.yaml [hunk 2]
```diff
@@ -0,0 +1,144 @@
+---
```
Comment (dirgim) [inline syncer710.yaml:1]: Is this file needed here or can it be removed?

Comment (hongweiliu17) [reply syncer710.yaml:1]: No, it is not needed, when I use it, I have a namespace creation part above. I will remove it. Thanks.

Comment (hongweiliu17) [reply syncer710.yaml:1]: This is an example generated by command `kubectl kcp workload sync mycluster` and I add some CRD into it.

**Review by dirgim — APPROVED** [review]
LGTM, just the `bin/controller-gen` needs to be resolved before merging
